src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -437,7 +437,15 @@ bool HandleRedirect (HttpStatusCode redirectCode, HttpURLConnection httpConnecti
 			if (redirectState.RedirectCounter >= MaxAutomaticRedirections)
 				throw new WebException ($"Maximum automatic redirections exceeded (allowed {MaxAutomaticRedirections}, redirected {redirectState.RedirectCounter} times)");
 
-			Uri location = new Uri (locationHeader [0], UriKind.Absolute);
+			string redirectUrl = locationHeader[0];
+			string protocol = httpConnection.URL?.Protocol;
+			if (redirectUrl.StartsWith("//") && protocol != null && protocol.StartsWith("http", StringComparison.OrdinalIgnoreCase))
```

Comment:
The first string comparison should be ordinal as well. And what about `https`? Won't this kind of redirection work in that case?